src/python/grpcio_tests/tests/interop/server_ssl_cert_reload_test/_test.py

```diff
@@ -0,0 +1,363 @@
+'''
+
+import os
+import unittest
+from concurrent import futures
+
+import grpc
+from src.proto.grpc.testing import test_pb2_grpc
+
+from tests.interop import methods
+
+
+DIR = os.path.abspath(os.path.dirname(__file__))
+
+def _get_abs_path(path):
+    return DIR + '/' + path
+
+
+ca_1_pem = open(_get_abs_path('cert_hier_1/certs/ca.cert.pem')).read()
+ca_2_pem = open(_get_abs_path('cert_hier_2/certs/ca.cert.pem')).read()
+
+client_key_1_pem = \
+    open(_get_abs_path('cert_hier_1/intermediate/private/client.key.pem')).read()
+client_chain_1_pem = '\n'.join([
+    open(_get_abs_path('cert_hier_1/intermediate/certs/client.cert.pem')).read(),
+    open(_get_abs_path('cert_hier_1/intermediate/certs/intermediate.cert.pem')).read(),
+    ])
+
+client_key_2_pem = \
+    open(_get_abs_path('cert_hier_2/intermediate/private/client.key.pem')).read()
+client_chain_2_pem = '\n'.join([
+    open(_get_abs_path('cert_hier_2/intermediate/certs/client.cert.pem')).read(),
+    open(_get_abs_path('cert_hier_2/intermediate/certs/intermediate.cert.pem')).read(),
+    ])
+
+server_key_1_pem = \
+    open(_get_abs_path('cert_hier_1/intermediate/private/localhost-1.key.pem')).read()
+server_cert_1_pem = '\n'.join([
+    open(_get_abs_path('cert_hier_1/intermediate/certs/localhost-1.cert.pem')).read(),
+    open(_get_abs_path('cert_hier_1/intermediate/certs/intermediate.cert.pem')).read(),
+])
+
+server_key_2_pem = \
+    open(_get_abs_path('cert_hier_2/intermediate/private/localhost-1.key.pem')).read()
+server_cert_2_pem = '\n'.join([
+    open(_get_abs_path('cert_hier_2/intermediate/certs/localhost-1.cert.pem')).read(),
+    open(_get_abs_path('cert_hier_2/intermediate/certs/intermediate.cert.pem')).read(),
+])
+
+
+class CertConfigFetcher(object):
+  def __init__(self, ca_1_pem, my_key_1_pem, my_cert_1_pem,
+               ca_2_pem, my_key_2_pem, my_cert_2_pem,
+               switch_cert_on_client_num):
+    self.client_num = -1
+    self.ca_2_pem = ca_2_pem
+    self.my_key_2_pem = my_key_2_pem
+    self.my_cert_2_pem = my_cert_2_pem
+    self.switch_cert_on_client_num = switch_cert_on_client_num
+    self.cert_config = grpc.ssl_server_certificate_config(
+        [(my_key_1_pem, my_cert_1_pem)],
+        root_certificates=ca_1_pem,
+      )
+    # we simulate cb failures at 2 and 3
+    assert self.switch_cert_on_client_num > 3
+
+  def __call__(self):
+    self.client_num += 1
+    if self.client_num == 0:
+      # this is actually during server initialzation
+      return self.cert_config
+    else:
+      if self.client_num in (2, 3):
+        raise ValueError('just an error for fun, should not affect the test')
+
+      if self.client_num != self.switch_cert_on_client_num:
+        return None
+      else:
+        # do the switch
+        return grpc.ssl_server_certificate_config(
+          [(self.my_key_2_pem, self.my_cert_2_pem)],
+          root_certificates=self.ca_2_pem,
+        )
+
+
+class _ServerSSLCertReloadTestBase(unittest.TestCase):
```

Comment:
Don't put `Base` in the names of classes you intend to use as base classes.

Reply:
will update

Reply:
done